src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -3326,7 +3327,26 @@ public final T blockLast(Duration timeout) {
 	 * the same key (yet element keys can repeat in the overall sequence)
 	 */
 	public final <V> Flux<T> distinctUntilChanged(Function<? super T, ? extends V> keySelector) {
-		return onAssembly(new FluxDistinctUntilChanged<>(this, keySelector));
+		Objects.requireNonNull(keySelector, "keySelector");
+		Function<? super T, ? extends V> nullSafeKeySelector =
+				keySelector.andThen(key -> Objects.requireNonNull(key,
```

Comment:
should be done inside the target operator if possible keeping the keySelector argument as before. Rx does keep both too